src/core/src/pass/serialize.cpp

```diff
@@ -1206,6 +1206,19 @@ bool pass::Serialize::run_on_model(const std::shared_ptr<ov::Model>& model) {
     if (m_xmlFile && m_binFile) {
         serializeFunc(*m_xmlFile, *m_binFile, model, m_version);
     } else {
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+        auto xmlPath_wstr = ov::util::string_to_wstring(m_xmlPath);
```

Comment:
can we re-write it this way?
```
#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
        bool is_uni = true;
        const auto& xmlPath_ref = ov::util::string_to_wstring(m_xmlPath);
        const auto& binPath_ref = ov::util::string_to_wstring(m_binPath);
#else 
        bool is_uni = false;
        const auto& xmlPath_ref = m_xmlPath;
        const auto& binPath_ref = m_binPath;
#endif
...
...
OPENVINO_ASSERT(xml_file, "Can't open xml file "  + (is_uni ? "." : xmlPath_ref ));
```

Reply:
`OPENVINO_ASSERT(xml_file, "Can't open xml file "  + (is_uni ? "." : xmlPath_ref ));`
fails on Win as wstring and string can't be concatenated:
`no conversion from 'const std::wstring' to 'const char [2]'`

We can change to:
`OPENVINO_ASSERT(bin_file, L"Can't open bin file: " + binPath_ref ));`

But OPENVINO_ASSERT still fails for wstring:
```
C:\Users\popovaan\git\openvino\src\core\include\openvino/core/except.hpp(40,1): error C2679: binary '<<': no operator f
ound which takes a right-hand operand of type 'T' (or there is no acceptable conversion) [C:\Users\popovaan\git\openvin
o\build\src\core\openvino_core_obj.vcxproj]
```

Reply:
Discussed offline, please check the changes.